### PR TITLE
null fields for email - phone, check during create and update if it a…

### PR DESCRIPTION
**Pull request by Bozhokmaria**
…lready exists

### CODE REVIEW

File: src/main/java/com/exadel/discountwebapp/vendor/repository/VendorRepository.java [hunk 1]
```diff
@@ -19,6 +19,8 @@
 
     Optional<Vendor> findByEmail(String email);
 
+    Optional<Vendor> findByPhoneNumber(String email);
```
Comment (ahlukhouski2) [inline src/main/java/com/exadel/discountwebapp/vendor/repository/VendorRepository.java]: ...(String email) ?

File: src/main/java/com/exadel/discountwebapp/vendor/service/VendorService.java [hunk 2]
```diff
@@ -45,7 +45,8 @@ public VendorResponseVO findByTitle(String title) {
 
     @Transactional(propagation = Propagation.REQUIRED)
     public VendorResponseVO create(VendorRequestVO request) {
-        vendorEmailValidator.validate(request);
+        vendorValidator.checkDuplicateEmail(request);
+        vendorValidator.checkDuplicatePhones(request);
```
Comment (ahlukhouski2) [inline src/main/java/com/exadel/discountwebapp/vendor/service/VendorService.java]: these 2 validations can be performed inside of one "vendorValidator.validateForCreate(request)" method

File: src/main/java/com/exadel/discountwebapp/vendor/service/VendorService.java [hunk 3]
```diff
@@ -54,9 +55,8 @@ public VendorResponseVO update(Long id, VendorRequestVO request) {
         Vendor vendor = getVendorById(id);
         String imageUrl = vendor.getImageUrl();
 
-        if (!vendor.getEmail().equals(request.getEmail())) {
-            vendorEmailValidator.validate(request);
-        }
+        vendorValidator.checkDuplicateEmailForUpdate(vendor, request);
+        vendorValidator.checkDuplicatePhonesForUpdate(vendor, request);
```
Comment (ahlukhouski2) [inline src/main/java/com/exadel/discountwebapp/vendor/service/VendorService.java]: same here

File: src/main/java/com/exadel/discountwebapp/vendor/repository/VendorRepository.java (hunk 1 above)
Comment (Bozhokmaria) [reply src/main/java/com/exadel/discountwebapp/vendor/repository/VendorRepository.java]: corrected, my mistake

File: src/main/java/com/exadel/discountwebapp/vendor/service/VendorService.java (hunk 2 above)
Comment (Bozhokmaria) [reply src/main/java/com/exadel/discountwebapp/vendor/service/VendorService.java]: done

File: src/main/java/com/exadel/discountwebapp/vendor/service/VendorService.java (hunk 3 above)
Comment (Bozhokmaria) [reply src/main/java/com/exadel/discountwebapp/vendor/service/VendorService.java]: done

**Review by ahlukhouski2 — APPROVED** [review]

**Review by VolodymyrBushko — APPROVED** [review]

**Pull request merged**